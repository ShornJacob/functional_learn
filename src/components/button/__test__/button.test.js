import React from 'react'
import { render, cleanup } from '@testing-library/react';
import Button from '../button'
import "@testing-library/jest-dom/extend-expect"
import renderer from "react-test-renderer"

afterEach(cleanup)

test("renders button correctly", ()=> {
    const { getByTestId } = render(<Button label="Click me"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Click me")
})

test("renders button correctly", ()=> {
    const { getByTestId } = render(<Button label="save"></Button>)
    expect(getByTestId('button')).toHaveTextContent("save")
})

test("mastchessnapshot 1", () => {
    const tree = renderer.create(<Button label="save"></Button>).toJSON()
    expect(tree).toMatchSnapshot()
})

test("mastchessnapshot 2", () => {
    const tree = renderer.create(<Button label="save me"></Button>).toJSON()
    expect(tree).toMatchSnapshot()
})