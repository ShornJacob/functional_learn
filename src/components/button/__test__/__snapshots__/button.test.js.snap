// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mastchessnapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;

exports[`mastchessnapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save me
</div>
`;
